src/nvim/CMakeLists.txt

```diff
@@ -144,7 +151,7 @@ if(CLANG_ASAN_UBSAN OR CLANG_MSAN OR CLANG_TSAN)
 endif()
 
 get_directory_property(gen_includes INCLUDE_DIRECTORIES)
-foreach(gen_include ${gen_includes})
+foreach(gen_include ${gen_includes} ${LUAJIT_INCLUDE_DIRS})
```

Comment:
BTW, theoretically this may cause problems if LuaJIT uses macros for defining `lua_…` types and makes them different from lua ones as a result and PREFER_LUAJIT is false. I do not see this actually happening and AFAIK I am currently using onle the one lua type in function signatures: `lua_State`.

This is left as-is because I am either using luajit or both lua and luajit and I do not want to generate two versions of files with declarations.